### ISSUE TITLE
Else branch with conditional let allows undefined variables

### ISSUE THREAD

**Issue by SheepTester** [nbuilding/N-lang#75]
Minimal example:

```rust
if let <yes test> = none {
	print(test)
} else {
	print(test)
}
```

This causes a runtime error:

```
...
  File "...\N-lang\python\scope.py", line 162, in get_
variable
    raise NameError("You tried to get a variable/function `%s`, but it isn't def
ined." % name)
NameError: You tried to get a variable/function `test`, but it isn't defined.
```

I suspect that the type checker uses the same scope for both the if and else branch, so the variables bound by the conditional let seem to be available in the else branch by the type checker.

**Comment by Ashvin-Ranjan** [comment]
Hmm, sounds like we just... separate the scopes

**Comment by SheepTester** [comment]
probably

**Comment by Ashvin-Ranjan** [comment]
Fixed as of PR #86
